src/values/AbstractObjectValue.js

```diff
@@ -543,7 +543,7 @@ export default class AbstractObjectValue extends AbstractValue {
       let d2 = ob2.$GetOwnProperty(P);
       let d2val =
         d2 === undefined ? this.$Realm.intrinsics.undefined : IsDataDescriptor(this.$Realm, d2) ? d2.value : undefined;
-      if (d1val === undefined || d2val === undefined) {
+      if (d1val === undefined || d2val === undefined || (d1 === undefined && d2 === undefined)) {
```

Comment:
I mentioned this in my summary. In the cases where `d1` and `d2` are both undefined, it's not really safe to continue to then do `Join.joinValuesAsConditional(this.$Realm, cond, d1val, d2val);` as one of the React tests broke when we did this (bad output).

Reply:
If neither of the two objects has the property, perhaps we could just return undefined?

Reply:
We can't unfortunately, I tried that as my first approach. The issue is that the property might be on the prototype (which is ironically what they are in this case).